components/compose/awesomebar/src/main/java/mozilla/components/compose/browser/awesomebar/internal/Suggestions.kt

```diff
@@ -36,30 +35,25 @@ internal fun Suggestions(
         state = state,
         modifier = Modifier.testTag("mozac.awesomebar.suggestions")
     ) {
-        val suggestions = fetcher.state.value
```

Comment:
I think not having this `fetcher` that would update the data to display as a property inside of LazyColumn that would also recompose when the data changes would be the most important improvement.

Reply:
I am not sure I fully understand this change :). Don't need to block on it, but curious if you could share more?

Reply:
The update to a newer compose version definitely played a part but I would also put it on the fact that here we have a recomposition loop (not bad bad but bad enough - https://developer.android.com/jetpack/compose/phases#recomp-loop) and the fact that composable functions can run in parallel - https://developer.android.com/jetpack/compose/mental-model#parallel.

Not sure what synchronization mechanisms are used to prevent issue with running composables in parallel but just speculating that it may be possible that because `fetcher.state.value` is read in `Suggestions` it may happen that any change in the value will trigger:
- a recomposition of the items shown (the composables which adapts those values)
- this will trigger a recomposition of the parent  - `Suggestions`
- `Suggestions` will again create a property to hold `fetcher.state.value`. (We should've used `remember{}` to prevent this)
- the items shouldn't recompose since the actual value didn't change but maybe it did in the meantime.

Maybe these updates happen in less than a frame and recompositions didn't have time to handle it all, maybe the states are updated and a new composition is driven by different cores which may have out of sync values.
There are many "maybe"s but by moving that code for observing state out of the composable that should display the state the code becomes cleaner, more in line with Compose's direction of passing state down to be displayed - https://developer.android.com/jetpack/compose/state and hopefully skips some recompositions or at least leads a more direct line of updates, without that loop I talked about above.